pkg/fswatcher/fswatcher.go

```diff
@@ -4,41 +4,72 @@
 package fswatcher
 
 import (
-	"errors"
-	"fmt"
+	"hash/fnv"
+	"io"
 	"os"
-	"path/filepath"
+	"sync"
+	"testing"
 
-	"github.com/fsnotify/fsnotify"
-	"github.com/sirupsen/logrus"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const (
+	// how often tracked targets are checked for changes by default
+	defaultInterval = 5 * time.Second
+
+	// when fswatcher detects that it runs in a test, it will refresh things much faster
+	testInterval = 50 * time.Millisecond
+)
+
+// Event closely resembles what fsnotify.Event provided
+type Event struct {
+	// Path to the file or directory.
+	//
+	// Paths are relative to the input; for example with Add("dir") the Name
+	// will be set to "dir/file" if you create that file, but if you use
+	// Add("/path/to/dir") it will be "/path/to/dir/file".
+	Name string
+
+	// File operation that triggered the event.
+	//
+	// This is a bitmask and some systems may send multiple operations at once.
+	// Use the Event.Has() method instead of comparing with ==.
+	Op Op
+}
+
+// Op describes a set of file operations.
+type Op uint32
```

Comment:
unless we have a good reason, let Go decide of the length
```suggestion
type Op uint
```